prepare.sh

```diff
@@ -1,7 +1,64 @@
-#!/bin/sh
+#!/bin/bash
 
 set -x
 
 yum -y install git podman
 
-git clone --single-branch --branch=master https://github.com/ceph/ceph-csi /opt/build
+ARGUMENT_LIST=(
+    "ref"
+    "workdir"
+    "gitrepo"
+)
+
+opts=$(getopt \
+    --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}") $(printf "help::")" \
+    --name "$(basename "${0}")" \
+    --options "" \
+    -- "$@"
+)
+
+eval set --$opts
```

Comment:
IIRC there should be a space between the "--" and "$opts". Plus, it should probably be "${opts}". I highly encourage all vars to use quotes ("") and `${...}` in all locations (unless there's a good reason not to).